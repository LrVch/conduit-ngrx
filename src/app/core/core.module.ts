import { NgModule, Optional, SkipSelf } from '@angular/core';
import { CommonModule } from '@angular/common';
import { HTTP_INTERCEPTORS } from '@angular/common/http';
import { HttpTokenInterceptor } from './interceptors/http.token.interceptor';

import {
  ApiService,
  ArticlesService,
  AuthGuard,
  CommentsService,
  JwtService,
  ProfilesService,
  TagsService,
  UserService,
  CustomEventManager,
  ScrollService,
  DomUtilService
} from './services';
import { EventManager } from '@angular/platform-browser';

@NgModule({
  imports: [
    CommonModule
  ],
  providers: [
    { provide: HTTP_INTERCEPTORS, useClass: HttpTokenInterceptor, multi: true },
    ApiService,
    ArticlesService,
    AuthGuard,
    CommentsService,
    JwtService,
    ProfilesService,
    TagsService,
    UserService,
    ScrollService,
    DomUtilService,
    { provide: EventManager, useClass: CustomEventManager }
  ],
  declarations: []
})
export class CoreModule {
  constructor (@Optional() @SkipSelf() parentModule: CoreModule) {
    if (parentModule) {
      throw new Error(
        'CoreModule is already loaded. Import it in the AppModule only');
    }
  }
 }
