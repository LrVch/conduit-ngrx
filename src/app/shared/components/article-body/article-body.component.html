<div [innerHTML]="body | markdown"></div>
