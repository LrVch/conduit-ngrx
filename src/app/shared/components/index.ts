export * from './article/article.component';
export * from './articles-list/articles-list.component';
export * from './content-tabs/content-tabs.component';
export * from './tags/tags.component';
export * from './favorite/favorite.component';
export * from './layout';
export * from './list-error/list-error.component';
export * from './user-info/user-info.component';
export * from './follow-button/follow-button.component';
export * from './tag-list/tag-list.component';
export * from './comments/comments.component';
export * from './base-from/base-from.component';
export * from './password-strength/password-strength.component';
export * from './confirm/confirm.component';
export * from './comment-form/comment-form.component';
export * from './article-user/article-user.component';
export * from './article-full/article-full.component';
export * from './comment/comment.component';
export * from './article-body/article-body.component';
export * from './paginator/paginator.component';
export * from './dummy-article-item/dummy-article-item.component';
export * from './dummy-tags/dummy-tags.component';
export * from './delete-btn/delete-btn.component';
export * from './image-validator/image-validator.component';
