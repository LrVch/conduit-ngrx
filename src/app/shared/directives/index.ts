export * from './avatar.directive';
export * from './accent-on-invalid-from-field.directive';
