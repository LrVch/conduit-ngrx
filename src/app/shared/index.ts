export * from './article/article.component';
export * from './articles-list/articles-list.component';
export * from './content-tabs/content-tabs.component';
export * from './tags/tags.component';
export * from './favorite/favorite.component';
export * from './layout';
export * from './shared.module';
export * from './utils';
export * from './list-error/list-error.component';
export * from './show-authed.directive';
export * from './user-info/user-info.component';
export * from './follow-button/follow-button.component';
export * from './tag-list/tag-list.component';
export * from './comments/comments.component';
export * from './base-from/base-from.component';
