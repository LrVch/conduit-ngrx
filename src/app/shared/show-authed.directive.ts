import {
    Directive,
    Input,
    OnInit,
    TemplateRef,
    ViewContainerRef
} from '@angular/core';

import { AppState } from '../reducers';
import { Store, select } from '@ngrx/store';
import { selectAuthLoggedIn } from '../auth/auth.selectors';

@Directive({ selector: '[appShowAuthed]' })
export class ShowAuthedDirective implements OnInit {
    constructor(
        private templateRef: TemplateRef<any>,
        private viewContainer: ViewContainerRef,
        private store: Store<AppState>
    ) { }

    condition: boolean;

    ngOnInit() {
        this.store.pipe(select(selectAuthLoggedIn)).subscribe(
            (isAuthenticated) => {
                if (isAuthenticated && this.condition || !isAuthenticated && !this.condition) {
                    this.viewContainer.createEmbeddedView(this.templateRef);
                } else {
                    this.viewContainer.clear();
                }
            }
        );
    }

    @Input() set appShowAuthed(condition: boolean) {
        this.condition = condition;
    }

}
