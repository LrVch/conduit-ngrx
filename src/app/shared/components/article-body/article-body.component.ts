import { Component, Input, ViewEncapsulation } from '@angular/core';

@Component({
  selector: 'app-article-body',
  templateUrl: './article-body.component.html',
  styleUrls: ['./article-body.component.scss'],
  encapsulation: ViewEncapsulation.None
})
export class ArticleBodyComponent {
  @Input('body') body: string = '' as string;
}
