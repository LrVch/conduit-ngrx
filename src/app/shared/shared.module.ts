import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { HttpClientModule } from '@angular/common/http';
import { RouterModule } from '@angular/router';
import { MaterialModule } from './material/material.module';
import { ListErrorComponent } from './list-error/list-error.component';
import { TagsComponent } from './tags/tags.component';
import { ArticlesListComponent } from './articles-list/articles-list.component';
import { ContentTabsComponent } from './content-tabs/content-tabs.component';
import { ArticleComponent } from './article/article.component';
import { FavoriteComponent } from './favorite/favorite.component';
import { ShowAuthedDirective } from './show-authed.directive';
import { PageNotFoundComponent } from './layout/page-not-found/page-not-found.component';
import { UserInfoComponent } from './user-info/user-info.component';
import { FollowButtonComponent } from './follow-button/follow-button.component';
import { TagListComponent } from './tag-list/tag-list.component';
import { CommentsComponent } from './comments/comments.component';
import { CommentComponent } from './comment/comment.component';
import { BaseFromComponent } from './base-from/base-from.component';
import { AccentOnInvalidFromFieldDirective } from './directives';
import { PasswordStrengthComponent } from './password-strength/password-strength.component';
import { ConfirmComponent } from './confirm/confirm.component';
import { AvatarDirective } from './directives';
import { CommentFormComponent } from './comment-form/comment-form.component';

const components = [
  ListErrorComponent,
  TagsComponent,
  ArticlesListComponent,
  ContentTabsComponent,
  ArticleComponent,
  FavoriteComponent,
  ShowAuthedDirective,
  PageNotFoundComponent,
  UserInfoComponent,
  FollowButtonComponent,
  TagListComponent,
  CommentsComponent,
  CommentComponent,
  BaseFromComponent,
  AccentOnInvalidFromFieldDirective,
  PasswordStrengthComponent,
  ConfirmComponent,
  AvatarDirective,
  CommentFormComponent
];

const modules = [
  CommonModule,
  FormsModule,
  ReactiveFormsModule,
  HttpClientModule,
  RouterModule,
  MaterialModule
];


@NgModule({
  imports: modules,
  declarations: components,
  exports: [
    ...modules,
    ...components
  ],
  entryComponents: [ConfirmComponent],
})
export class SharedModule { }
