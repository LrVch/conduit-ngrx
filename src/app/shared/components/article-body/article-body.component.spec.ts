import { async, ComponentFixture } from '@angular/core/testing';
import { ConfigureFn, configureTests } from '@app/lib/testing';
import { ArticleBodyComponent } from './article-body.component';
import { MarkdownPipe } from '@app/shared/pipes';

describe('ArticleBodyComponent', () => {
  let component: ArticleBodyComponent;
  let fixture: ComponentFixture<ArticleBodyComponent>;

  beforeEach(async(() => {
    const configure: ConfigureFn = testBed => {
      testBed.configureTestingModule({
        declarations: [ArticleBodyComponent, MarkdownPipe]
      });
    };

    configureTests(configure).then(testBed => {
      fixture = testBed.createComponent(ArticleBodyComponent);
      component = fixture.componentInstance;
    });
  }));

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should has "body" @Input', () => {
    expect(component.body).toBe('');
  });

  it('should compile', () => {
    component.body = 'body';
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });
});
