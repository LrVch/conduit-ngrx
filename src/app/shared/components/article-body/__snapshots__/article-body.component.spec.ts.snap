// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleBodyComponent should compile 1`] = `
<app-article-body
  body={[Function String]}
>
  <div>
    <p>
      body
    </p>
    
  
  </div>
</app-article-body>
`;
